alf/utils/tensorrt_utils.py

```diff
@@ -344,6 +348,19 @@ def __init__(self,
                 a tuple of args.
             example_kwargs: The example kwargs to be used for ``method``. Should
                 be a dict of kwargs.
+            engine_file: if provided, this class will first check if such a file
+                exists. If so, it will load the engine from the file and skip the
+                build process. If not, the built engine will be saved to this file.
+                When a file is provided, it is the *user's responsibility* to ensure
+                that the loaded engine will work correctly in the current context;
+                no check will be performed by this class.
+                NOTE: This option is mainly for an inter-process cache and its
+                priority is lower than the in-process engine caching mechanism.
+                Within a process when the same compiled method is called multiple
+                times with the same signature, engine file loading and ``force_build_engine``
+                won't have any effect.
```

Comment:
by "won't have any effect" do you mean it will always build the engine?

Reply:
> by "won't have any effect" do you mean it will always build the engine?

whether it will or not depends on how the in-process cache mechanism. There is always a cache in the process memory.